### PR TITLE
CI: exclude docs requirements changes

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Exclude building test of the change is only to a template or other docs building setup
see: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-and-excluding-paths

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @carmocca @akihironitta @borda

### CODE REVIEW

File: .github/workflows/ci-app-examples.yml [hunk 1]
```diff
@@ -12,7 +12,8 @@ on:
       - "src/lightning_app/**"
       - "tests/tests_app_examples/**"
       - "examples/app_*"
-      - "requirements/app/**"
+      - "requirements/app/*"
+      - "!requirements/app/docs.txt"
```
Comment (carmocca) [inline .github/workflows/ci-app-examples.yml:16]: I'm not sure that this works as you expect.

A change to `src/foo/bar` matches `!requirements/app/docs.txt`.
Whereas I believe you want to just exclude this particular file.

You have two options.

a) List all `.txt` files explicitly here without `docs.txt`
b) Use paths-ignore: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-excluding-paths

Same comment applies to all other changes

**Review by carmocca — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-app-examples.yml (hunk 1 above)
Comment (Borda) [reply .github/workflows/ci-app-examples.yml:16]: I believe that if it does not work, it would be a bug on their side... :confused:

Comment (carmocca) [reply .github/workflows/ci-app-examples.yml:16]: Oops I misread GitHub docs. Resolving

File: .github/workflows/ci-app-examples.yml [hunk 2]
```diff
@@ -12,7 +12,8 @@ on:
       - "src/lightning_app/**"
       - "tests/tests_app_examples/**"
       - "examples/app_*"
-      - "requirements/app/**"
+      - "requirements/app/*"
```
Comment (carmocca) [inline .github/workflows/ci-app-examples.yml:15]: Why remove the **?

File: .github/workflows/ci-pkg-install.yml [hunk 3]
```diff
@@ -12,6 +12,7 @@ on:
       - "setup.py"
       - "src/**"
       - "requirements/**"
+      - "!requirements/*/docs.txt"
```
Comment (carmocca) [inline .github/workflows/ci-pkg-install.yml:15]: Shouldn't this have 2 `*`?

Comment (Borda) [reply .github/workflows/ci-pkg-install.yml:15]: so far I know `**` is for any depth and `*` is just one folder

File: .github/workflows/ci-app-examples.yml (hunk 2 above)
Comment (Borda) [reply .github/workflows/ci-app-examples.yml:15]: I think that the depth is just 1 not any

File: .github/workflows/ci-pkg-install.yml (hunk 3 above)
Comment (carmocca) [reply .github/workflows/ci-pkg-install.yml:15]: But in the line above you use `**`

Comment (Borda) [reply .github/workflows/ci-pkg-install.yml:15]: i could be wrong, but... 
- `**` would cover files in requirements and all subfolders
- `*` would cover only files directly in requirements but no subfolder

Comment (Borda) [reply .github/workflows/ci-pkg-install.yml:15]: I believe that this is the same pattern: https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/util/pattern/PathPattern.html

**Review by carmocca — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

**Comment by Borda** [comment]
@carmocca reverted :)

**Review by carmocca — APPROVED** [review]
LGTM again. Needs https://github.com/Lightning-AI/lightning/pull/15503 to be merged first

**Pull request merged**